### PR TITLE
Spec: Pagination

### LINKED ISSUE #76: Paginate the specification
The specification could benefit from some pagination. Could we keep a table of contents and a DEX overview where the spec is, but break off some of the more technical sections to their own mediawiki files? The "chapters", each with its own file, could roughly follow the current table of contents.

1. Communication Protocols
2. Exchange Fundamentals
3. Account Management & Community Conduct
4. Data API (including order book subscriptions)
5. Order Management
6. Exchange Administration

The specification is bound to grow in size. I think this would provide some sanity to the information.

### PR DESCRIPTION
**Pull request by JoeGruffins**
Depends upon #63, #83, #84
Closes #76  

Current pages:
 - Abstract
 - Communication Protocols
 - Distributed Exchange Design Fundamentals
 - Distributed Exchange Administration
 - Client Accounts
 - Client Order Management
 - Data API
 - Atomic Settlement
 - Community Conduct

### CODE REVIEW

**Comment by JoeGruffins** [comment]
The spec is currently treated as the "DRAFT specification"
Is this still accurate? Perhaps the original draft specification should be saved separately with a link, while the actual specification grows and changes.

So, the Data API, is it just the https api? Should the client ws commands be there too?

**Comment by buck54321** [comment]
Specification should stay in draft for now. Data API will need both websocket and https endpoints, but only websockets at first.

**Comment by buck54321** [comment]
@JoeGruffins Please rebase. We seem to always have a steady stream of spec updates rolling in, so the timing here is never going to be perfect. I say we move pagination up the priority list so we can have the framework to build on. Thoughts, @chappjc?

**Comment by JoeGruffins** [comment]
@buck54321 ok

File: spec/ORDERMANAGEMENT.mediawiki [hunk 1]
```diff
@@ -0,0 +1,889 @@
...
+* [[#Market_Order|Market orders]] are used to buy or sell a specified amount of an asset at the best price available.
+* [[#Cancel_Order|Cancel orders]] are used to remove standing limit orders from the order book.
+
+The order book holds only limit orders with time in force ''standing'' that have
+not completely filled or been cancelled. All other orders are only valid for one
+[[#Epochbased_Order_Matching|epoch match cycle]].
+
+===Session Authentication===
+
+Orders must be placed on an authenticated connection. Once a websocket
```
Comment (buck54321) [inline spec/ORDERMANAGEMENT.mediawiki]: Let's move this section to **communications protocols** and change this first sentence to read

> Many DEX messages must be sent on an authenticated connection.

File: spec/ATOMICSETTLEMENT.mediawiki [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+<img src="images/ab-profiles.png" align="right">
+# The lot size for the market is 1 DCR.
+# All backing funds for the orders are in the form of a single coin.
+# There are no fees.
+
+The examples involve a participant named Alice, who holds some DCR and would
+like to exchange it for some BTC.
+The market is a DCR/BTC market, where DCR is the $B!H(Bbase asset$B!I(B and BTC is the
+$B!H(Bquote asset$B!I(B.
```
Comment (buck54321) [inline spec/ATOMICSETTLEMENT.mediawiki]: Hwat? 

These characters are all over this file.

File: spec/README.mediawiki [hunk 3]
```diff
@@ -79,1681 +79,42 @@ basis for design.
...
 describes the notable design aspects that facilitate an exchange service with
 the features described above.
 
-'''[[#Client_Order_Management|Client Order Management]]''' details the different
+'''[[EXCHANGEADMIN.mediawiki|Distributed Exchange Administration]]''' describes
+the tasks required to administer the exchange.
+
+'''[[ACCOUNTS.mediawiki|Client Accounts]]''' details account creation.
+
+'''[[ORDERMANAGEMENT.mediawiki|Client Order Management]]''' details the different
```
Comment (buck54321) [inline spec/README.mediawiki]: What do you think about numbering the "chapters"?

File: spec/EXCHANGEFUNDAMENTALS.mediawiki [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+As the interface is expected to evolve greatly during initial implementation,
+the specifics of the interface structure will be released at a later time.
+
+With the exception of Decred and Bitcoin, which will be implemented during
+initial phases of DEX development, it is expected that development communities
+will release their own appropriately vetted plugins.
+
+==References==
+
+&#91;7&#93; [https://github.com/satoshilabs/slips/blob/master/slip-0044.md BIP-0044 Registered Coins
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply spec/EXCHANGEFUNDAMENTALS.mediawiki]: LOL. wtf

File: spec/ATOMICSETTLEMENT.mediawiki (hunk 2 above)
Comment (JoeGruffins) [reply spec/ATOMICSETTLEMENT.mediawiki]: They are double quotations and apostrophes.  They look fine in my editor... hmm...

File: spec/COMMPROTOCOLS.mediawiki [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+The primary reason for limiting the scope of the HTTP API is to eliminate client
+polling for rapidly changing resources, such as order status API endpoints.
+
+==References==
+
+&#91;4&#93; [https://tools.ietf.org/html/rfc6455 The WebSocket Protocol]
+
+&#91;5&#93; [https://tools.ietf.org/html/rfc8446 Transport Layer Security 1.3]
+
+&#91;6&#93; [https://tools.ietf.org/html/rfc8259 JSON]
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply spec/COMMPROTOCOLS.mediawiki]: add References

File: spec/README.mediawiki (hunk 3 above)
Comment (JoeGruffins) [reply spec/README.mediawiki]: Numbered with brackets

File: spec/ORDERMANAGEMENT.mediawiki (hunk 1 above)
Comment (JoeGruffins) [reply spec/ORDERMANAGEMENT.mediawiki]: looks ok?

**Comment by JoeGruffins** [comment]
ah, missed the filenames

**Comment by buck54321** [comment]
Many of the anchor links need to be updated as well.

File: spec/comm.mediawiki [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+If a client unexpectedly disconnects with active orders, the orders may match in
+the client's absence. A list of any pending matches is included in the response.
+
+<code>result</code>
+{|
+! field      !! type   !! description
+|-
+| startepoch || int   || the epoch at which trading did or will commence. May be in the future e.g. [[#Trade_Suspension|after maintenance]]
+|-
+| matches    || &#91;object&#93; || list of [[#Match_notifications|Match objects]]
```
Comment (JoeGruffins) [inline spec/comm.mediawiki]: this link broken in master

File: spec/orders.mediawiki [hunk 7]
```diff
@@ -0,0 +1,839 @@
...
+
+'''Request route:''' <code>init</code>, '''originator:''' client
+
+<code>payload</code>
+{|
+! field      !! type   !! description
+|-
+| orderid    || string || the order ID
+|-
+| matchid    || string    || the matchid, retrieved from the [[#Match_notifications|match notification]]
```
Comment (JoeGruffins) [inline spec/orders.mediawiki]: Same link broken in master

**Comment by JoeGruffins** [comment]
Where should I point #Match_notifications

**Comment by chappjc** [comment]
@JoeGruffins https://github.com/decred/dcrdex/commit/50b7228c28296d613cd8c8d4823a20c3da13adca#diff-4e063bde9bfc271b86c087a530bdbc35R1125
Looks correct to change the `#Match_notifications` link to `#Match_negotiation`

**Pull request merged**